### PR TITLE
Credential-less AWS/GCP/Azure Secrets Store support

### PR DESCRIPTION
**Pull request by stefannica**
## Describe changes
Allow the Secrets Stores to use the AWS/GCP/Azure Service Connector implicit authentication methods without enabling them globally. These authentication methods are by default disabled because they are insecure, but the Secrets Store can still use them internally to manage external secrets.

This PR also makes a couple of modifications to the GCP and AWS connectors to support credential-less implicit authentication when the ZenML server is running in AWS:

* allows the AWS implicit authentication method to be used with an IAM role. This will be the preferred configuration option for ZenML Cloud tenants that connect to a customer AWS Secrets Manager because it involves no exchange of credentials and the customer has complete control over who has access to these credentials.
* implements a workload identity authentication for the GCP connector, which allows the GCP connector to authenticate to GCP using AWS credentials.
  
## Pre-requisites
Please ensure you have done the following:
- [x] I have read the **CONTRIBUTING.md** document.
- [x] If my change requires a change to docs, I have updated the documentation accordingly.
- [ ] I have added tests to cover my changes.
- [x] I have based my new branch on `develop` and the open PR is targeting `develop`. If your branch wasn't based on develop read [Contribution guide on rebasing branch to develop](https://github.com/zenml-io/zenml/blob/main/CONTRIBUTING.md#-pull-requests-rebase-your-branch-on-develop).
- [ ] If my changes require changes to the dashboard, these changes are communicated/requested.

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] Other (add details above)

## Summary by CodeRabbit

- **New Features**
	- Updated AWS service connector documentation to include guidance on configuring optional IAM session policies and alternative authentication methods.
	- Enhanced AWS and Azure service connectors with improved authentication flows and options.
	- Enabled implicit authentication methods for service connectors, with specific support for AWS, Azure, and GCP.
- **Documentation**
	- Added comprehensive details on accessing EKS Kubernetes clusters and using IAM roles with implicit authentication in AWS.
- **Refactor**
	- Modified the control flow for Azure and GCP service connectors to refine the authentication process.
- **Chores**
	- Adjusted ZenML server deployment settings to support optional `role_arn` for implicit authentication methods.

### CODE REVIEW

**Comment by safoinme** [comment]
@coderabbitai review

File: src/zenml/integrations/aws/service_connectors/aws_service_connector.py [hunk 1]
```diff
@@ -241,13 +245,39 @@ class AWSAuthenticationMethods(StrEnum):
 need to be adjusted to allows listing and accessing/describing the AWS resources
 that the connector is configured to access.
 
+An IAM role may optionally be specified to be assumed by the connector on top of
+the implicit credentials. This is only possible when the implicit credentials
+have permissions to assume the target IAM role. Configuring an IAM role has all
+the advantages of the AWS IAM Role authentication method plus the added benefit
+of not requiring any explicit credentials to be configured and stored:
+
+* the connector will generate temporary STS tokens upon request by
```
Comment (safoinme) [inline src/zenml/integrations/aws/service_connectors/aws_service_connector.py:254]:
```suggestion
* The connector will generate temporary STS tokens upon request by
```

File: src/zenml/integrations/aws/service_connectors/aws_service_connector.py [hunk 2]
```diff
@@ -241,13 +245,39 @@ class AWSAuthenticationMethods(StrEnum):
...
+[calling the AssumeRole STS API](https://docs.aws.amazon.com/IAM/latest/UserGuide/id\_credentials\_temp\_request.html#api\_assumerole).
```
Comment (safoinme) [inline src/zenml/integrations/aws/service_connectors/aws_service_connector.py:255]:
```suggestion
[calling the AssumeRole STS API](https://docs.aws.amazon.com/IAM/latest/UserGuide/id_credentials_temp_request.html#api_assumerole).
```

**Review by safoinme — APPROVED** [review]
Looks good

**Review by AlexejPenner — APPROVED** [review]
:seal: I'm not very familiar with this part of the code, but what I understand looks good :)

File: src/zenml/integrations/azure/service_connectors/azure_service_connector.py [hunk 3]
```diff
@@ -1086,8 +1086,6 @@ def _auto_configure(
         expiration_seconds: Optional[int] = None
         expires_at: Optional[datetime.datetime] = None
         if auth_method == AzureAuthenticationMethods.IMPLICIT:
-            cls._check_implicit_auth_method_allowed()
-
             auth_config = AzureBaseConfig(
                 resource_group=resource_group,
                 storage_account=storage_account,
```
[the remark this answers is not in the document]
Comment (stefannica) [reply src/zenml/integrations/azure/service_connectors/azure_service_connector.py:1091]: while generally true, this check is only removed for methods that are executed locally client-side (i.e. auto-configuration and local client configuration), while the check is still in place server-side, where the security concern is really relevant.

File: src/zenml/integrations/gcp/service_connectors/gcp_service_connector.py [hunk 4]
```diff
@@ -1235,8 +1235,6 @@ def _auto_configure(
             )
 
         if auth_method == GCPAuthenticationMethods.IMPLICIT:
-            cls._check_implicit_auth_method_allowed()
-
             auth_config = GCPBaseConfig(
                 project_id=project_id,
             )
```
[the remark this answers is not in the document]
Comment (stefannica) [reply src/zenml/integrations/gcp/service_connectors/gcp_service_connector.py:1467]: while generally true, this check is only removed for methods that are executed locally client-side (i.e. auto-configuration and local client configuration), while the check is still in place server-side, where the security concern is really relevant.

**Pull request merged**